[ivalue] operator<<: don't error on invalid IValue tags

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112232

While running the profiler, we observed a scenario where we observe IValues with invalid tags. Specifically, we try to convert the IValue to a string here:

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/torch/csrc/profiler/util.cpp#L306-L308

and in the scenario with invalid IValues, an exception gets thrown here, in `operator<<`.

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/aten/src/ATen/core/ivalue.cpp#L864

IMO, `<<` shouldn't error if the IValue is bad; instead, we should just print that the IValue tag is invalid.

Differential Revision: [D50760040](https://our.internmc.facebook.com/intern/diff/D50760040)

davidberard98 committed:
[ivalue] operator<<: don't error on invalid IValue tags



Differential Revision: [D50701674](https://our.internmc.facebook.com/intern/diff/D50701674/)

[ghstack-poisoned]

davidberard98 committed:
Update on "[ivalue] operator<<: don't error on invalid IValue tags"


Differential Revision: [D50701674](https://our.internmc.facebook.com/intern/diff/D50701674/)

While running the profiler, we observed a scenario where we observe IValues with invalid tags. Specifically, we try to convert the IValue to a string here:

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/torch/csrc/profiler/util.cpp#L306-L308

and in the scenario with invalid IValues, an exception gets thrown here, in `operator<<`.

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/aten/src/ATen/core/ivalue.cpp#L864

IMO, `<<` shouldn't error if the IValue is bad; instead, we should just print that the IValue tag is invalid.

[ghstack-poisoned]

davidberard98 committed:
Update on "[ivalue] operator<<: don't error on invalid IValue tags"


Differential Revision: [D50701674](https://our.internmc.facebook.com/intern/diff/D50701674/)

While running the profiler, we observed a scenario where we observe IValues with invalid tags. Specifically, we try to convert the IValue to a string here:

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/torch/csrc/profiler/util.cpp#L306-L308

and in the scenario with invalid IValues, an exception gets thrown here, in `operator<<`.

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/aten/src/ATen/core/ivalue.cpp#L864

IMO, `<<` shouldn't error if the IValue is bad; instead, we should just print that the IValue tag is invalid.

[ghstack-poisoned]

albanD commented on aten/src/ATen/core/ivalue.cpp:
```diff
@@ -861,7 +861,7 @@ std::ostream& operator<<(std::ostream & out, const IValue & v) {
 
   }
-  AT_ERROR("Tag not found: ", v.tagKind());
+  return out << "InvalidIValue(tag=" << std::to_string(static_cast<uint32_t>(v.tag)) << ")";
```
nit: this is not consistent with other objects being printed here. But I guess that's fine.

davidberard98 replied:
good point - switched it to `<Invalid IValue tag=1234>`

davidberard98 committed:
Update on "[ivalue] operator<<: don't error on invalid IValue tags"


Differential Revision: [D50701674](https://our.internmc.facebook.com/intern/diff/D50701674/)

While running the profiler, we observed a scenario where we observe IValues with invalid tags. Specifically, we try to convert the IValue to a string here:

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/torch/csrc/profiler/util.cpp#L306-L308

and in the scenario with invalid IValues, an exception gets thrown here, in `operator<<`.

https://github.com/pytorch/pytorch/blob/d3bf6803b62c79f1dafd1eec49b4bd65d5a27697/aten/src/ATen/core/ivalue.cpp#L864

IMO, `<<` shouldn't error if the IValue is bad; instead, we should just print that the IValue tag is invalid.

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D50760040).

davidberard98 commented:
@pytorchbot merge

davidberard98 commented:
@pytorchbot merge
